tests/erlang_tests/floatmath.erl

```diff
@@ -0,0 +1,247 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2022 Fred Dushin <fred@dushin.net>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(floatmath).
+
+-export([start/0, id/1, to_int/1]).
+
+-define(PI_OVER_2, 1.570796325).
+-define(THREE_PI_OVER_2, 4.712388975).
+-define(PI_OVER_4, 0.7853981625).
+-define(THREE_QUARTERS_PI, 2.3561944875).
+
+start() ->
+    ok = test_sin(),
+    ok = test_cos(),
+    ok = test_tan(),
+    ok = test_asin(),
+    ok = test_acos(),
+    ok = test_atan(),
+
+    ok = test_sinh(),
+    ok = test_cosh(),
+    ok = test_tanh(),
+    ok = test_asinh(),
+    ok = test_acosh(),
+    ok = test_atanh(),
+
+    ok = test_floor(),
+    ok = test_ceil(),
+    ok = test_pow(),
+    ok = test_exp(),
+    ok = test_sqrt(),
+    ok = test_log(),
+    ok = test_log10(),
+    ok = test_log2(),
+    0.
+
+test_sin() ->
+    erlang:display(test_sin),
+    0 = to_int(math:sin(id(0))),
+    0 = to_int(math:sin(id(0.0))),
+    1 = to_int(math:sin(id(?PI_OVER_2))),
+    0 = to_int(math:sin(id(math:pi()))),
+    -1 = to_int(math:sin(id(?THREE_PI_OVER_2))),
+    ok.
+
+test_cos() ->
+    erlang:display(test_cos),
+    1 = to_int(math:cos(id(0))),
+    1 = to_int(math:cos(id(0.0))),
+    0 = to_int(math:cos(id(?PI_OVER_2))),
+    -1 = to_int(math:cos(id(math:pi()))),
+    0 = to_int(math:cos(id(?THREE_PI_OVER_2))),
+    ok.
+
+test_tan() ->
+    erlang:display(test_tan),
+    1 = to_int(math:tan(id(?PI_OVER_4))),
+    -1 = to_int(math:tan(id(?THREE_QUARTERS_PI))),
+    ok.
+
+test_asin() ->
+    erlang:display(test_asin),
+    0 = to_int(math:asin(id(0))),
+    0 = to_int(math:asin(id(0.0))),
+    badarg = catch_error(fun() -> to_int(math:asin(id(2.0))) end),
+    ok.
+
+test_acos() ->
+    erlang:display(test_acos),
+    2 = to_int(math:acos(id(0))),
+    2 = to_int(math:acos(id(0.0))),
+    badarg = catch_error(fun() -> to_int(math:acos(id(2.0))) end),
```

Comment:
I would favor a test that passes with Erlang/OTP and therefore this should be `badarith`.

Reply:
Done